File: examples/canonical-faker/scala/src/test/scala/io/agentcontextbase/faker/SmokeTest.scala
package io.agentcontextbase.faker

import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers

class SmokeTest extends AnyFlatSpec with Matchers {
  "Smoke profile" should "generate 3 organizations" in {
    val dataset = Pipeline.generate(Profile.SMOKE)
    dataset.organizations should have length 3
  }
  
  it should "pass validation" in {
    val dataset = Pipeline.generate(Profile.SMOKE)
    val report = Validate.check(dataset)
    report.ok shouldBe true
    report.violations shouldBe empty
  }
  
  it should "be reproducible" in {
    val d1 = Pipeline.generate(Profile.SMOKE)
    val d2 = Pipeline.generate(Profile.SMOKE)
    d1.organizations shouldBe d2.organizations
  }
}

File: examples/canonical-faker/elixir/test/tenant_core/smoke_test.exs
defmodule TenantCore.SmokeTest do
  use ExUnit.Case

  test "smoke profile generates 3 organizations" do
    dataset = TenantCore.Pipeline.generate(TenantCore.Profile.smoke())
    assert length(dataset.organizations) == 3
  end

  test "smoke profile passes validation" do
    dataset = TenantCore.Pipeline.generate(TenantCore.Profile.smoke())
    assert dataset.report.ok == true
    assert dataset.report.violations == []
  end

  test "smoke profile is reproducible" do
    d1 = TenantCore.Pipeline.generate(TenantCore.Profile.smoke())
    d2 = TenantCore.Pipeline.generate(TenantCore.Profile.smoke())
    assert d1.organizations == d2.organizations
  end

  test "smoke profile has at least one non-free organization (weighted distribution)" do
    dataset = TenantCore.Pipeline.generate(TenantCore.Profile.small())
    plans = Enum.map(dataset.organizations, & &1.plan)
    free_count = Enum.count(plans, &(&1 == "free"))
    # free should be roughly 50%; allow wider range for small N
    assert free_count < length(plans)
  end
end
